crates/nu-command/src/filesystem/cp.rs

```diff
@@ -154,6 +154,19 @@ impl Command for Cp {
                     if src.is_file() {
                         let res = if interactive && dst.exists() {
                             interactive_copy_file(interactive, src, dst, span)
+                        } else if src == dst {
```

Comment:
this `else if src == dst` should check first rather than `if interactive && dst.exists()`.  Or else `cp test_file test_file -i` will also clear the origin file

Reply:
Interesting, I thought `-i` is not in use, it turned out my nu is out of dated :) Thanks for the catching

Reply:
Done